trick_source/sim_services/MonteCarlo/MonteVarFile.cpp

```diff
@@ -30,64 +30,78 @@ std::string Trick::MonteVarFile::describe_variable()
     std::stringstream ss;
 
     ss << "#NAME:\t\t" << this->name << "\n"
-       << "#TYPE:\t\tFILE\n" 
+       << "#TYPE:\t\tFILE\n"
        << "#UNIT:\t\t" << this->unit << "\n"
        << "#FILE:\t\t" << this->file_name << "\n"
        << "#COLUMN:\t" << this->column << "\n";
 
     return ss.str();
 }
 
-std::string Trick::MonteVarFile::get_next_value() {
-    if (input_file_stream->good()) {
-        double file_value;
-        char *current_position = buffer;
-        do {
+std::string Trick::MonteVarFile::get_next_value()
+{
+    if (input_file_stream->good())
+    {
+        // Skip the comments and empty lines in the data file.
+        do
+        {
             input_file_stream->getline(buffer, 4096);
```

Comment:
What happens if there are more than 4096 characters in the line? According to [the documentation](http://www.cplusplus.com/reference/istream/istream/getline/), the first call to `getline` will get the first 4096 characters, and the next call will continue from there, not actually starting on the next line. Although there's some stuff about `sentry` objects at the end that I don't understand. Should probably test this. Either way, this appears to implicitly limit the number of columns that can be used.

Forget all that. Use [`std::getline`](http://en.cppreference.com/w/cpp/string/basic_string/getline) instead. Then you can read into a `std::string` that automatically sizes itself instead of a fixed-size `char` array. No more arbitrary limit. Bonus: it probably simplifies the rest of file. I know you didn't write the original code, and that's not really what your changes are about, but you might as well improve it while you're in here!

Reply:
+1 Sounds like a good idea. I'll make that change.